Source/WebCore/rendering/BreakLines.h

```diff
@@ -178,48 +182,57 @@ inline size_t BreakLines::nextBreakablePosition(CachedLineBreakIteratorFactory&
                 continue;
             }
 
+            // When nonBreakingSpaceBehavior == NoBreakSpaceBehavior::Break is specified, NBSP should behave as if it is just a space.
+            CharacterType beforeCharacter = before;
+            if constexpr (nonBreakingSpaceBehavior == NoBreakSpaceBehavior::Break) {
+                if (beforeCharacter == noBreakSpace)
+                    beforeCharacter = space;
+            }
+
             // If both characters are ASCII, use a lookup table for enhanced speed
             // and for compatibility with other browsers (see comments on lineBreakTable for details).
-            if (before <= lineBreakTable.lastCharacter && after <= lineBreakTable.lastCharacter) {
-                if (before >= lineBreakTable.firstCharacter && after >= lineBreakTable.firstCharacter) {
-                    if (lineBreakTable.unsafeLookup(before, after))
+            if (beforeCharacter <= lineBreakTable.lastCharacter && after <= lineBreakTable.lastCharacter) {
+                if (beforeCharacter >= lineBreakTable.firstCharacter && after >= lineBreakTable.firstCharacter) {
+                    if (lineBreakTable.unsafeLookup(beforeCharacter, after))
                         return i;
                 } // Else at least one is an ASCII control character; don't break.
                 continue;
```

Comment:
- When we have before=space, we don't need to do an ascii table lookup, we can just `continue;` and avoid the conditionals after it. (See that it hits the continue clause just below.)
- It's weird to capture the before=nbsp case and not after=nbsp.
- If you capture after=nbsp, you need to assign both the character and the class, so there shouldn't be a new `CharacterType beforeCharacter` variable.
- It's not clear to me that adding a conditional this high up for a relatively rare character is wise, given how sensitive this code seems to be to adding conditionals--this adds a new one for every single character in the document. Instead I would consider capturing this case where we detect kGL in the `// Non-ASCII rapid lookup.` section below, overwrite the CharacterInfo value, use a `goto` to jump back to the top of `// ASCII rapid lookup`. (Idk if that conforms to your guidelines, or if it's worth doing, but I think it's worth considering.)

Reply:
Sounds good. Will add kSP and handle things.